### PR TITLE
Update k8s dependencies to 1.24

### PR DESCRIPTION
**Pull request by creydr**
**Is this a BUG FIX or a FEATURE ?**:

> Uncomment only one, leave it on its own line:
>
> /kind bug
/kind enhancement

**What this PR does / why we need it**:
Kubernetes 1.24 released a few days ago. This PR updates the dependencies to have the latest updates & patches.

**Special notes for your reviewer**:

**Release note**:
```release-note
none
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -68,9 +68,6 @@ LOCAL_REGISTRY ?= registry:5000
 
 export MANIFESTS_DIR ?= build/_output/manifests
 
-# Produce CRDs that work back to Kubernetes 1.11 (no version conversion)
-CRD_OPTIONS ?= "crd:trivialVersions=true"
```
Comment (rhrazdil) [inline Makefile:72]: +1

**Review by rhrazdil — COMMENTED** [review]
Looks like kubernetes 1.24 ditches `node-role.kubernetes.io/master`
We should substitute this label with `node-role.kubernetes.io/control-plane`.

Not sure if this PR is the right place for it, since it only updates go deps, but not the k8s cluster the CI is still running on.

**Comment by rhrazdil** [comment]
I'd say it's okay to substitute tha label later on when we update the k8s cluster in kubevirtci.
/lgtm

**Comment by qinqon** [comment]
/hold 
@creydr we should first bump kubevirtci to 1.24

**Comment by creydr** [comment]
@rhrazdil @qinqon could you recheck?

**Comment by creydr** [comment]
I added a commit (53ccebb13d7e7e62cc14b6748fe158663a47abcd) to respect the new `node-role.kubernetes.io/control-plane` label too, which seem to be now in (kubevirtci) 1.24.

**Comment by rhrazdil** [comment]
Seems like the updated affinities is affecting this SNO test?
https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/nmstate_kubernetes-nmstate/1084/pull-kubernetes-nmstate-e2e-operator-k8s/1537114413834702848

**Comment by creydr** [comment]
> Seems like the updated affinities is affecting this SNO test? https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/nmstate_kubernetes-nmstate/1084/pull-kubernetes-nmstate-e2e-operator-k8s/1537114413834702848

@rhrazdil Thanks for bringing this up. I addressed it in 2b3481c27c800846c0a81bed5b16000b6d846c1d

**Comment by creydr** [comment]
@rhrazdil @qinqon could you PTAL?
Thanks in advance

/retest

**Comment by qinqon** [comment]
/lgtm
If CI is fine we are good to go 

@rhrazdil can you approve in case it's ok ?

**Comment by creydr** [comment]
I think the upgrade job is failing, because the operator manifests from v0.71.0 do not respect the new `node-role.kubernetes.io/control-plane` label (still uses `node-role.kubernetes.io/master`) and so the pods can't get scheduled on the nodes. So we would need to backport something like 1ce168013 and 2b3481c27 to the v0.71 tag.

**Comment by qinqon** [comment]
> I think the upgrade job is failing, because the operator manifests from v0.71.0 do not respect the new `node-role.kubernetes.io/control-plane` label (still uses `node-role.kubernetes.io/master`) and so the pods can't get scheduled on the nodes. So we would need to backport something like [1ce1680](https://github.com/nmstate/kubernetes-nmstate/commit/1ce168013269ebaa16efa319afa886206032896f) and [2b3481c](https://github.com/nmstate/kubernetes-nmstate/commit/2b3481c27c800846c0a81bed5b16000b6d846c1d) to the v0.71 tag.

ok, backporting it.

**Comment by qinqon** [comment]
> > I think the upgrade job is failing, because the operator manifests from v0.71.0 do not respect the new `node-role.kubernetes.io/control-plane` label (still uses `node-role.kubernetes.io/master`) and so the pods can't get scheduled on the nodes. So we would need to backport something like [1ce1680](https://github.com/nmstate/kubernetes-nmstate/commit/1ce168013269ebaa16efa319afa886206032896f) and [2b3481c](https://github.com/nmstate/kubernetes-nmstate/commit/2b3481c27c800846c0a81bed5b16000b6d846c1d) to the v0.71 tag.
> 
> ok, backporting it.

@creydr can you do the backport since you are the commits owner.

**Comment by creydr** [comment]
@qinqon can you create a release branch for v0.71 to which I could backport the fixes? Or what is the usual workflow?

**Comment by qinqon** [comment]
> > @creydr can you do the backport since you are the commits owner.
> 
> @qinqon can you create a release branch for v0.71 to which I could backport the fixes? Or what is the usual workflow?

We should merge to main and release v0.72.0, there is no release branch yet.

**Comment by creydr** [comment]
> We should merge to main and release v0.72.0, there is no release branch yet.

:+1: 
https://github.com/nmstate/kubernetes-nmstate/pull/1103

**Comment by creydr** [comment]
> We should merge to main and release v0.72.0, there is no release branch yet.

@rhrazdil @qinqon as #1103 merged, could we do a new release so the update tests use the "correct" previous release?

**Comment by creydr** [comment]
@qinqon could you PTAL again? :)

**Pull request merged**